synfig-studio/src/gui/app.cpp

```diff
@@ -3731,15 +3687,37 @@ App::window_shortcuts()
 
 void App::open_img_in_external(const std::string &uri)
 {
-	synfig::info("Opening with external tool: " + uri);
-	if(!try_open_img_external(uri))
-	{
-		Gtk::MessageDialog dialog(*App::main_window, _("Make sure Preferred editing tool was set in \n Edit->Preferences->Editing:"), false, Gtk::MESSAGE_ERROR, Gtk::BUTTONS_CLOSE, true);
-		dialog.set_secondary_text(uri);
+	const std::string filename = Glib::filename_from_uri(uri);
+	synfig::info("Opening with external tool: " + filename);
+
+	// Check if external editor is set
+	if(App::image_editor_path.empty())	{
+		Gtk::MessageDialog dialog(*App::main_window, _("Please set preferred editing tool in \nEdit->Preferences->Editing"), false, Gtk::MESSAGE_ERROR, Gtk::BUTTONS_CLOSE, true);
+		dialog.set_secondary_text(filename);
 		dialog.set_title(_("Error"));
 		dialog.run();
+		return;
 	}
 
+	char buffer[4096];
+#ifdef WIN32
+	::snprintf(buffer, sizeof(buffer), "\"%s\" \"%s\"",App::image_editor_path.c_str(), filename.c_str());
```

Comment:
Why don't you use etl:strprintf?

Reply:
Agree. This need to be fixed.

Reply:
I don't know if it has that annoying problems with non-latin characters as it is.